server.go

```diff
@@ -66,18 +76,65 @@ func satisfiesBasicAuth(r *http.Request, user, password string) bool {
 	return expectedVal == foundVal
 }
 
-func getHandlerForBlockList(blockListCfg BlockListConfig) func(http.ResponseWriter, *http.Request) {
+func toValidCIDR(ip string) string {
+	if strings.Contains(ip, "/") {
+		return ip
+	}
+
+	if strings.Contains(ip, ":") {
+		return ip + "/128"
+	}
+	return ip + "/32"
+}
+
+func getTrustedIPs(ips []string) ([]net.IPNet, error) {
+	trustedIPs := make([]net.IPNet, 0)
+	for _, ip := range ips {
+		cidr := toValidCIDR(ip)
+		_, ipNet, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, err
+		}
+		trustedIPs = append(trustedIPs, *ipNet)
+	}
+	return trustedIPs, nil
+}
+
+func networksContainIP(networks []net.IPNet, ip string) bool {
+	parsedIP := net.ParseIP(ip)
+	for _, network := range networks {
+		if network.Contains(parsedIP) {
+			return true
+		}
+	}
+	return false
+}
+
+func getHandlerForBlockList(blockListCfg BlockListConfig) (func(http.ResponseWriter, *http.Request), error) {
+	trustedIPs, err := getTrustedIPs(blockListCfg.Authentication.TrustedIPs)
+	if err != nil {
+		return nil, err
+	}
 	f := func(w http.ResponseWriter, r *http.Request) {
-		apiCountersByFormatName[blockListCfg.Format].Inc()
-		if strings.EqualFold(blockListCfg.Authentication.Type, "basic") {
-			if !satisfiesBasicAuth(r, blockListCfg.Authentication.User, blockListCfg.Authentication.Password) {
-				http.Error(w, "access denied", http.StatusForbidden)
-				return
+		RouteHits.WithLabelValues(
+			blockListCfg.Endpoint,
+		).Inc()
+
+		ip, _, _ := net.SplitHostPort(r.RemoteAddr)
+		if !networksContainIP(trustedIPs, ip) {
```

Comment:
to make the logic more trivial, don't we want to have an type of auth like "ip_based" ?

Reply:
done